projects/storefrontlib/src/shared/components/form/file-upload/file-upload.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { FormControl } from '@angular/forms';
+
+/**
+ * Component that adds a file upload control.
+ */
+@Component({
+  selector: 'cx-file-upload',
+  templateUrl: './file-upload.component.html',
+  // we cannot use onPush change detection as the form state isn't updated without explicit
+  // change detection, see https://github.com/angular/angular/issues/10816
+})
+export class FileUploadComponent {
+  selectedFile: File;
+
+  @Input() control: FormControl;
```

Comment:
As this will be re-usable component can we add comments for inputs and methods with usage instruction?